cirq/google/engine/calibration.py

```diff
@@ -45,14 +46,23 @@ class Calibration(abc.Mapping):
 
         `calibration['t1']`
 
+    This class can be instantiated either from a `MetricsSnapshot` proto
+    or from a dictionary of metric values.
+
     Attributes:
         timestamp: The time that this calibration was run, in milliseconds since
             the epoch.
     """
 
-    def __init__(self, calibration: v2.metrics_pb2.MetricsSnapshot) -> None:
+    def __init__(self,
+                 calibration: v2.metrics_pb2.MetricsSnapshot = v2.metrics_pb2.
+                 MetricsSnapshot(),
+                 metrics: Optional[
+                     Dict[str, Dict[Tuple['cirq.GridQubit', ...], Any]]] = None
+                ) -> None:
         self.timestamp = calibration.timestamp_ms
-        self._metric_dict = self._compute_metric_dict(calibration.metrics)
+        self._metric_dict = (metrics or
+                             self._compute_metric_dict(calibration.metrics))
```

Comment:
IIUC, the intention is to compute the metrics if `None` is passed, however this also computes the metrics when empty dict is passed. We should only have one value that triggers such behavior, so the following is better:

```python
self._metric_dict = self._compute_metric_dict(calibration.metrics) if metrics is None else metrics
```